src/include/duckdb/optimizer/cardinality_estimator.hpp

```diff
@@ -0,0 +1,62 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/optimizer/table_filter_stats.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "duckdb/catalog/catalog_entry/table_catalog_entry.hpp"
+#include "duckdb/optimizer/join_node.hpp"
+#include "duckdb/planner/column_binding.hpp"
+#include "duckdb/planner/column_binding_map.hpp"
+#include "duckdb/planner/filter/conjunction_filter.hpp"
+#include "duckdb/planner/filter/constant_filter.hpp"
+
+namespace duckdb {
+
+static constexpr double DEFAULT_SELECTIVITY = 0.2;
+
+class CardinalityEstimator {
+public:
+	//! A mapping of base table index -> all columns used to determine the join order
+	unordered_map<idx_t, unordered_set<idx_t>> relation_to_columns;
+	//! A mapping of (relation, bound_column) -> (actual table, actual column)
+	column_binding_map_t<ColumnBinding> relation_column_to_original_column;
+	vector<column_binding_set_t> equivalent_relations;
+	vector<idx_t> equivalent_relations_tdom_no_hll;
+	vector<idx_t> equivalent_relations_tdom_hll;
+	unordered_map<idx_t, std::string> relation_to_table_name;
```

Comment:
Can we merge `relation_to_columns` and `relation_to_table_name`?

For example:
```c++
struct MyTable { // not sure about name ...
	string name;
	unordered_set<idx_t> column_ids;
}

// now we can have one map instead of two
unordered_map<idx_t, MyTable> relation_to_table;
```

This makes it easier to add things later.